markdownx/tests/tests.py

```diff
@@ -1,8 +1,35 @@
 from django.test import TestCase
+try:
+    # Django >= 2.0
+    from django.urls import reverse
+except ImportError:
+    from django.core.urlresolvers import reverse
+import os
+import re
 
 
 class SimpleTest(TestCase):
 
     def test_me(self):
         response = self.client.get('/testview/')
         self.assertEqual(response.status_code, 200)
+
+    def test_upload(self):
+        url = reverse('markdownx_upload')
+
+        with open('markdownx/tests/static/django-markdownx-preview.png', 'rb') as fp:
+            response = self.client.post(url, {'image': fp}, HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+
+            try:
+                # Django >= 1.9
+                json = response.json()
+            except AttributeError:
+                import json
+                json = json.loads(response.content.decode('utf-8'))
+
+            self.assertEqual(response.status_code, 200)
+            self.assertTrue('image_code' in json)
+
+            match = re.findall(r'^\!\[\]\(([\w\\\/\-\.]+)\)', json['image_code'])
```

Comment:
Maybe there's a better way to do this. The regex looks a bit too complicated. Also, it should be in try/finally or something so cleanup happens even if the test exits early.

Is `markdownx/15a2f283-a906-4d30-a461-61e1d6abfa20.png` meant to be added in this PR?